engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java

```diff
@@ -137,6 +141,30 @@ private boolean isValidProcess(
     return true;
   }
 
+  private boolean hasValidStartInstructions(
+      final CommandControl<ProcessInstanceCreationRecord> controller,
+      final ExecutableProcess process,
+      final ArrayProperty<ProcessInstanceCreationStartInstruction> startInstructions) {
+
+    return startInstructions.stream()
+        .map(ProcessInstanceCreationStartInstruction::getElementId)
+        .filter(elementId -> !isElementOfProcess(process, elementId))
+        .findAny()
+        .map(
+            elementId -> {
+              controller.reject(
+                  RejectionType.INVALID_ARGUMENT,
+                  "Expected to create instance of process with start instructions but no element found with id '%s'."
+                      .formatted(elementId));
```

Comment:
:thinking: Yes. In general, it would be nice to report all validation failures at once. 

But it would lead to more complicated code and I don't see much value for this use case.